deployment/stacks/topics.ts

```diff
@@ -0,0 +1,106 @@
+import { Construct } from "constructs";
+import {
+  aws_lambda as lambda,
+  aws_iam as iam,
+  aws_ec2 as ec2,
+  aws_ssm as ssm,
+  Duration,
+  Tags,
+  Aws,
+} from "aws-cdk-lib";
+import { Lambda } from "../constructs/lambda";
+
+interface CreateTopicsComponentsProps {
+  scope: Construct;
+  stage: string;
+  project: string;
+  vpc: ec2.IVpc;
+  kafkaAuthorizedSubnets: ec2.ISubnet[];
+  iamPermissionsBoundary: iam.IManagedPolicy;
+  iamPath: string;
+  brokerString: string;
+}
+
+export function createTopicsComponents(props: CreateTopicsComponentsProps) {
+  const {
+    scope,
+    stage,
+    project,
+    vpc,
+    kafkaAuthorizedSubnets,
+    iamPermissionsBoundary,
+    iamPath,
+    brokerString,
+  } = props;
+
+  const service = "topics";
+
+  const deleteTopicsEnabled =
+    stage === "main" || stage === "val" || stage === "production";
+
+  const lambdaSecurityGroup = new ec2.SecurityGroup(
+    scope,
+    "LambdaSecurityGroup",
+    {
+      vpc,
+      description: "Security Group for the topics service lambdas",
+      allowAllOutbound: true,
+    }
+  );
+
+  const commonProps = {
+    brokerString,
+    stackName: `${service}-${stage}`,
+    environment: {
+      brokerString,
+      project,
+    },
+    additionalPolicies: [
+      new iam.PolicyStatement({
+        effect: iam.Effect.ALLOW,
+        actions: ["ssm:GetParameter"],
+        resources: [
+          `arn:aws:ssm:${Aws.REGION}:${Aws.ACCOUNT_ID}:parameter/configuration/${stage}/*`,
+        ],
+      }),
+    ],
+    iamPermissionsBoundary,
+    iamPath,
+    vpc,
+    vpcSubnets: { subnets: kafkaAuthorizedSubnets },
+    securityGroups: [lambdaSecurityGroup],
+  };
+
+  const createTopicsLambda = new Lambda(scope, "CreateTopics", {
+    entry: "services/topics/handlers/createTopics.js",
+    handler: "handler",
+    timeout: Duration.seconds(60),
+    ...commonProps,
+    environment: { topicNamespace: "", ...commonProps.environment },
+  });
+
+  const deleteTopicsLambda = new Lambda(scope, "DeleteTopics", {
+    entry: "services/topics/handlers/deleteTopics.js",
+    handler: "handler",
+    timeout: Duration.seconds(300),
+    ...commonProps,
+  });
+
+  if (!deleteTopicsEnabled) {
```

Comment:
Based on this code I expected `deleteTopicsLambda`  to be inside this if.

Reply:
that makes sense

Reply:
Fixed, thanks!